hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerRatisRequest.java

```diff
@@ -58,14 +61,29 @@ public void testRequestWithNonExistentBucket()
 
     String volumeName = "vol1";
     String bucketName = "invalidBuck";
+
+    // Add entry to Volume Table.
+    omMetadataManager.getVolumeTable().addCacheEntry(
+        new CacheKey<>(omMetadataManager.getVolumeKey(volumeName)),
+        new CacheValue<>(
+            Optional.of(
+                OmVolumeArgs.newBuilder()
+                    .setVolume(volumeName)
+                    .setOwnerName("owner")
+                    .setAdminName("admin")
+                    .build()), 100L));
+
     OzoneManagerProtocolProtos.OMRequest omRequest = OMRequestTestUtils
         .createCompleteMPURequest(volumeName, bucketName, "mpuKey", "mpuKeyID",
             new ArrayList<>());
 
-    OMClientRequest req =
-        OzoneManagerRatisUtils.createClientRequest(omRequest, ozoneManager);
-    Assert.assertNotNull(req);
-    Assert.assertTrue("Unexpected request on invalid bucket",
-        req instanceof S3MultipartUploadCompleteRequest);
+    try {
+      // Request creation flow should throw exception if the bucket associated
+      // with the request doesn't exist.
+      OzoneManagerRatisUtils.createClientRequest(omRequest, ozoneManager);
+      fail("Expected OMException: Bucket not found");
+    } catch (OMException oe) {
+      // Expected exception.
```

Comment:
Can you add an assertion inside the catch block.

      // Expected exception
      Assert.assertEquals(OMException.ResultCodes.BUCKET_NOT_FOUND,
          oe.getResult());

Reply:
Thank you for the suggestion, I've added this in my latest commit.